RACK-Ontology/ontology/SOFTWARE.sadl

```diff
@@ -114,17 +117,25 @@ NAMESPACE
     is a COMPONENT_TYPE
     has identifier "NAMESPACE".
 
-COMPONENT
-    (note "An abstract software component")
+SWCOMPONENT
```

Comment:
I'm imaging software component graphs being used to build up AST-level representations of software. Do you think that SWCOMPONENT might become a subtype of SYSTEM?

Currently we define a SYSTEM as a logical component and I think of software components as the concrete components. It seems like there should be *some* distinction though I'm not certain what it should be yet.

Reply:
I would consider a SWCOMPONENT to be subtype of SYSTEM, but I am not sure I would define this subtype in the base ontology, this distinction seems like it should be done with the supplemental ontology.  The reason I say this is I could see various ways that you may decompose the software, AST-level, package-level, etc.  These different breakdowns may need some differing properties. Rather than trying to add all all the possible permutations in the base ontology we should have the skeletal structure and then things can be elaborated as needed in the supplemental.

Reply:
I'm not sure I agree one way or the other with supplemental vs. primary. Actually, I'd love to see NO supplementals, and a single data model, if that's borne out by the amount of overlap between performers.

However, I really do want to see a distinction between the logical hierarchy embodied in SYSTEM and the reified hierarchy embodied in SWCOMPONENT. We have relationships between them to show what logical parts are implemented by what software implementation parts, so there's no ambiguity or disconnection. But it's key (and expected by at least 2 of the TA1s) that the two structures be distinct.